kms/keys.js

```diff
@@ -0,0 +1,1467 @@
+  // const location = 'global';
+
+  // The name of the key ring, e.g. "my-key-ring"
+  // const keyRingName = 'my-key-ring';
+
+  // Builds and authorizes a Cloud KMS client
+  buildAndAuthorizeService((err, cloudkms) => {
+    if (err) {
+      console.log(err);
+      return;
+    }
+
+    const request = {
+      // This will be a path parameter in the request URL
+      resource_: `projects/${projectId}/locations/${location}/keyRings/${keyRingName}`
+    };
+
+    // Gets the IAM policy of a key ring
+    cloudkms.projects.locations.keyRings.getIamPolicy(request, (err, policy) => {
+      if (err) {
+        console.log(err);
+        return;
+      }
+
+      if (policy.bindings) {
+        policy.bindings.forEach((binding) => {
+          if (binding.members && binding.members.length) {
+            console.log(`${binding.role}:`);
+            binding.members.forEach((member) => {
+              console.log(`  ${member}`);
+            });
+          }
+        });
+      } else {
+        console.log(`Policy for key ring ${keyRingName} is empty.`);
+      }
+    });
+  });
+  // [END kms_get_keyring_policy]
+}
+
+function addMemberToKeyRingPolicy (projectId, location, keyRingName, member, role) {
+  // [START kms_add_member_to_keyring_policy]
+  // Your Google Cloud Platform project ID
+  // const projectId = 'YOUR_PROJECT_ID';
+
+  // The location of the key ring, e.g. "global"
+  // const location = 'global';
+
+  // The name of the key ring, e.g. "my-key-ring"
+  // const keyRingName = 'my-key-ring';
+
+  // The member to add to the key ring, e.g. "user:developer@company.com"
+  // const member = 'user:developer@company.com';
+
+  // The role to give the member, e.g. "roles/viewer"
+  // const role = 'roles/viewer';
+
+  // Builds and authorizes a Cloud KMS client
+  buildAndAuthorizeService((err, cloudkms) => {
+    if (err) {
+      console.log(err);
+      return;
+    }
+
+    let request = {
+      // This will be a path parameter in the request URL
+      resource_: `projects/${projectId}/locations/${location}/keyRings/${keyRingName}`
+    };
+
+    // Gets the IAM policy of a key ring
+    cloudkms.projects.locations.keyRings.getIamPolicy(request, (err, policy) => {
+      if (err) {
+        console.log(err);
+        return;
+      }
+
+      policy = Object.assign({ bindings: [] }, policy);
+
+      const index = policy.bindings.findIndex((binding) => binding.role === role);
+
+      // Add the role/member combo to the policy
+      const binding = Object.assign({
+        role: role,
+        members: []
+      }, policy.bindings[index]);
+      if (index === -1) {
+        policy.bindings.push(binding);
+      }
+      if (!binding.members.includes(member)) {
+        binding.members.push(member);
+      }
+
+      request = {
+        // This will be a path parameter in the request URL
+        resource_: `projects/${projectId}/locations/${location}/keyRings/${keyRingName}`,
```

Comment:
Wishful thinking: these parameter names (`resource_` and `resource`) are annoyingly similar. (Not that we can really change that... 🙁 )